scripts/apps/authoring/authoring/directives/ItemCarouselDirective.ts

```diff
@@ -111,29 +110,35 @@ export function ItemCarouselDirective(notify) {
                     waitForMediaToLoad(mediaItems).then(initCarousel);
                 });
             });
+
+            function getItemsCount(items: Array<any>): number {
+                return Object.values(items)
```

Comment:
Why are you calling `Object.values` on an array?

`getItemsCount` function is pure and doesn't use any state from `ItemCarouselDirective`. Can you move it out of the directive and place it in the root file scope? It's easier to understand the code that way, because when you see a function in some scope, but it doesn't really use anything from that scope, you need to spend time to realize that. On the other hand, when you put it in the root scope, it's impossible for that function to use anything from another scope, so you don't even have to check.

Reply:
> place it in the root file scope?

Which file do you mean?

Reply:
Same file `ItemCarouselDirective.ts`. Move the function to around line 34.